### PR TITLE
fix(dashboard-tests): Fix timeout exceeded failures

### PR DESCRIPTION
**Pull request by shruthilayaj**
The `'can enter edit mode for widgets'` test was failing
with timeouts on CI.

So it seemed like the async openModal calls on the other 
tests in this module had lingering side effects which were 
causing the timeouts. Removed the asyncModal stuff in 
favour of spying on the openModal function to ensure the 
modal openend. We lose some coverage which ensured 
that the PUT/POST mocks were called with the right 
payload. For now, enhanced the modal tests to make 
sure the "onSubmit" on these modals are called with 
the right info.

### CODE REVIEW

**Comment by chadwhitacre** [comment]
@shruthilayaj Thoughts on https://github.com/getsentry/sentry/pull/30501 as a bandaid? :)

**Comment by chadwhitacre** [comment]
Per https://github.com/getsentry/sentry/issues/30500#issuecomment-988951981, would it make sense to enable `--verbose` on this PR and compare the timings of these tests to a local baseline? Based on my local results it seems like these tests should not be taking nearly as long as they probably are. My hypothesis is that _all_ of the cases in these test suites are running waaaaaay longer in CI than local. Is that a valid hypothesis? :)

```
 PASS  tests/js/spec/views/dashboardsV2/detail.spec.jsx (7.255 s)
  Dashboards > Detail
    prebuilt dashboards
      ✓ can delete (334 ms)
      ✓ can rename and save (132 ms)
      ✓ disables buttons based on features (45 ms)
    custom dashboards
      ✓ can remove widgets (278 ms)
      ✓ can enter edit mode for widgets (660 ms) <------------------------------------- vs. 5000+
      ✓ does not update if api update fails (252 ms)
      ✓ shows add wiget option (426 ms)
      ✓ hides add widget option (227 ms)
      ✓ hides and shows breadcrumbs based on feature (271 ms)
      ✓ enters edit mode when given a new widget in location query (113 ms)
      ✓ enters view mode when not given a new widget in location query (107 ms)
      ✓ can add library widgets (309 ms)
      ✓ disables add library widgets when max widgets reached (73 ms)
      ✓ adds an Issue widget to the dashboard (416 ms)

```

```
 PASS  tests/js/spec/views/alerts/create.spec.jsx (9.255 s)
  ProjectAlertsCreate
    ✓ redirects to wizard (439 ms)
    Issue Alert
      ✓ loads default values (111 ms)
      ✓ can remove filters, conditions and actions (1615 ms) <------------------------- vs. 5000+
      ✓ updates values and saves (1403 ms)
```

**Comment by chadwhitacre** [comment]
# `detail.spec.jsx`

## Local

```
 PASS  tests/js/spec/views/dashboardsV2/detail.spec.jsx (7.255 s)
  Dashboards > Detail
    prebuilt dashboards
      ✓ can delete (334 ms)
      ✓ can rename and save (132 ms)
      ✓ disables buttons based on features (45 ms)
    custom dashboards
      ✓ can remove widgets (278 ms)
      ✓ can enter edit mode for widgets (660 ms)
      ✓ does not update if api update fails (252 ms)
      ✓ shows add wiget option (426 ms)
      ✓ hides add widget option (227 ms)
      ✓ hides and shows breadcrumbs based on feature (271 ms)
      ✓ enters edit mode when given a new widget in location query (113 ms)
      ✓ enters view mode when not given a new widget in location query (107 ms)
      ✓ can add library widgets (309 ms)
      ✓ disables add library widgets when max widgets reached (73 ms)
      ✓ adds an Issue widget to the dashboard (416 ms)

```

## Remote

```
PASS tests/js/spec/views/dashboardsV2/detail.spec.jsx (15.794 s)
  Dashboards > Detail
    prebuilt dashboards
      ✓ can delete (656 ms)
      ✓ can rename and save (519 ms)
      ✓ disables buttons based on features (84 ms)
     custom dashboards
      ✓ can remove widgets (379 ms)
      ✓ can enter edit mode for widgets (5023 ms)
      ✓ does not update if api update fails (441 ms)
      ✓ shows add wiget option (482 ms)
      ✓ hides add widget option (438 ms)
      ✓ hides and shows breadcrumbs based on feature (501 ms)
      ✓ enters edit mode when given a new widget in location query (211 ms)
      ✓ enters view mode when not given a new widget in location query (199 ms)
      ✓ can add library widgets (674 ms)
      ✓ disables add library widgets when max widgets reached (132 ms)
      ✓ adds an Issue widget to the dashboard (2234 ms)
```

https://github.com/getsentry/sentry/runs/4460623804?check_suite_focus=true#step:9:212

# `create.spec.jsx`

## Local

```
 PASS  tests/js/spec/views/alerts/create.spec.jsx (9.255 s)
  ProjectAlertsCreate
    ✓ redirects to wizard (439 ms)
    Issue Alert
      ✓ loads default values (111 ms)
      ✓ can remove filters, conditions and actions (1615 ms)
      ✓ updates values and saves (1403 ms)
```

## Remote

```
PASS tests/js/spec/views/alerts/create.spec.jsx (13.929 s)
  ProjectAlertsCreate
    ✓ redirects to wizard (419 ms)
    Issue Alert
      ✓ loads default values (188 ms)
      ✓ can remove filters, conditions and actions (3439 ms)
      ✓ updates values and saves (2834 ms)
```

https://github.com/getsentry/sentry/runs/4460623804?check_suite_focus=true#step:9:1105

**Comment by chadwhitacre** [comment]
Hypothesis disproven! Eyeballing these results, CI looks to be roughly 2x local on average. For the `can remove filters` case this is enough to push it past 5 s on occasion, it being long-running on local to begin with. But! For the `can enter edit mode` case the runs here are showing more like 8x CI over local! Is there some extra db traffic in that case that exacerbates the difference? If so it may be worthwhile to tune that one case, though at a hair over 5 seconds we could probably also get away with leaving it as-is, given the higher timeout in #30501.

File: tests/js/spec/views/dashboardsV2/detail.spec.jsx [hunk 1]
```diff
@@ -359,18 +365,31 @@ describe('Dashboards > Detail', function () {
...
+                conditions: 'event.type:error',
+                fields: ['count()'],
+                name: '',
+              },
+            ],
+            title: 'Errors',
+            type: 'line',
+          },
+        })
+      );
```
Comment (shruthilayaj) [inline tests/js/spec/views/dashboardsV2/detail.spec.jsx:392]: Hmm, this change significantly reduced the run time of this test locally (~800ms -> 350ms) but made no difference on CI.

Local
```  Dashboards > Detail
    prebuilt dashboards
      ✓ can delete (325 ms)
      ✓ can rename and save (133 ms)
      ✓ disables buttons based on features (41 ms)
    custom dashboards
      ✓ can remove widgets (268 ms)
      ✓ can enter edit mode for widgets (378 ms)  <---
      ✓ does not update if api update fails (406 ms)
      ✓ shows add wiget option (187 ms)
      ✓ hides add widget option (162 ms)
      ✓ hides and shows breadcrumbs based on feature (206 ms)
      ✓ enters edit mode when given a new widget in location query (102 ms)
      ✓ enters view mode when not given a new widget in location query (94 ms)
      ✓ can add library widgets (289 ms)
      ✓ disables add library widgets when max widgets reached (76 ms)
      ✓ adds an Issue widget to the dashboard (560 ms)
```
CI
```
PASS tests/js/spec/views/dashboardsV2/detail.spec.jsx (17.832 s)
  Dashboards > Detail
    prebuilt dashboards
      ✓ can delete (909 ms)
      ✓ can rename and save (289 ms)
      ✓ disables buttons based on features (71 ms)
    custom dashboards
      ✓ can remove widgets (474 ms)
      ✓ can enter edit mode for widgets (5571 ms)  <--- still over 5000 ms!
      ✓ does not update if api update fails (752 ms)
      ✓ shows add wiget option (380 ms)
      ✓ hides add widget option (353 ms)
      ✓ hides and shows breadcrumbs based on feature (429 ms)
      ✓ enters edit mode when given a new widget in location query (214 ms)
      ✓ enters view mode when not given a new widget in location query (195 ms)
      ✓ can add library widgets (718 ms)
      ✓ disables add library widgets when max widgets reached (159 ms)
      ✓ adds an Issue widget to the dashboard (2079 ms)
```

**Comment by shruthilayaj** [comment]
So it seemed like the async openModal calls on the other tests in this module had lingering side effects which were causing the timeouts. Removed the asyncModal stuff in favour of spying on the openModal function to ensure the modal openend. We lose some coverage which ensured that the PUT/POST mocks were called with the right payload. For now, enhanced the modal tests to make sure the "onSubmit" on these modals are called with the right info.

from CI logs:
```
PASS tests/js/spec/views/dashboardsV2/detail.spec.jsx (6.762 s)
  Dashboards > Detail
    prebuilt dashboards
      ✓ can delete (1085 ms)
      ✓ can rename and save (413 ms)
      ✓ disables buttons based on features (64 ms)
    custom dashboards
      ✓ can remove widgets (347 ms)
      ✓ opens edit modal for widgets (353 ms)
      ✓ does not update if api update fails (605 ms)
      ✓ shows add wiget option (319 ms)
      ✓ hides add widget option (292 ms)
      ✓ hides and shows breadcrumbs based on feature (358 ms)
      ✓ enters edit mode when given a new widget in location query (180 ms)
      ✓ enters view mode when not given a new widget in location query (163 ms)
      ✓ can add library widgets (187 ms)
      ✓ disables add library widgets when max widgets reached (159 ms)
      ✓ adds an Issue widget to the dashboard (137 ms)
```

Going to remove the verbose flag now.

**Review by chadwhitacre — APPROVED** [review]
> opens edit modal for widgets (353 ms)

I'd call that a win! 👏 🌮 @shruthilayaj 

Seems reasonable to me but I'm not deep in js testing every day, so take my approval with a grain of salt. ;-)

**Review by edwardgou-sentry — APPROVED** [review]
LGTM

**Pull request merged**